### PR TITLE
Node diver metrics

### PR DESCRIPTION
**Pull request by batphonghan**

### CODE REVIEW

File: shared/services/beacon/client/std-http-client.go [hunk 1]
```diff
@@ -572,6 +584,21 @@ func (c *StandardHttpClient) getSyncStatus() (SyncStatusResponse, error) {
...
+func (c *StandardHttpClient) getNodeVersion() (NodeVersionResponse, error) {
+	responseBody, status, err := c.getRequest(RequestNodeVersionPath)
+	if err != nil {
+		return NodeVersionResponse{}, fmt.Errorf("Could not get node sync status: %w", err)
+	}
+	if status != http.StatusOK {
+		return NodeVersionResponse{}, fmt.Errorf("Could not get node sync status: HTTP status %d; response body: '%s'", status, string(responseBody))
+	}
+	var nodeVersion NodeVersionResponse
+	if err := json.Unmarshal(responseBody, &nodeVersion); err != nil {
```
Comment (bharath-123) [inline shared/services/beacon/client/std-http-client.go:596]: can you refactor this code? abstract it to the http client

File: shared/services/ec-manager.go [hunk 2]
```diff
@@ -520,3 +523,51 @@ func (p *ExecutionClientManager) runFunction(function ecFunction) (interface{},
...
+	client := &http.Client{}
+	req, err := http.NewRequest(method, url, payload)
+
+	if err != nil {
+		return "", err
+	}
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := client.Do(req)
+	if err != nil {
```
Comment (bharath-123) [inline shared/services/ec-manager.go:553]: Please refactor this code too.

File: shared/services/ec-manager.go [hunk 3]
```diff
@@ -520,3 +523,51 @@ func (p *ExecutionClientManager) runFunction(function ecFunction) (interface{},
...
+	if !p.primaryReady && !p.fallbackReady {
+		return "", fmt.Errorf("EC not ready")
+	}
+
+	var url string
+
+	if p.primaryReady {
+		url = p.primaryEcUrl
+	} else {
+		url = p.primaryEcUrl
```
Comment (bharath-123) [inline shared/services/ec-manager.go:537]: shouldn't this be the fallback url?

File: stader/node/node.go [hunk 4]
```diff
@@ -385,12 +401,166 @@ func run(c *cli.Context) error {
...
+		cfg, err := services.GetConfig(c)
+		if err != nil {
+			errorLog.Printlnf("Error getconfig %+v", err)
+			return
+		}
+
+		for {
+			infoLog.Printlnf("Running the node diversity tracker daemon")
+
+			message, err := makesNodeDiversityMessage(ec, bc, pnr, w, cfg)
```
Comment (bharath-123) [inline stader/node/node.go:422]:
```suggestion
			message, err := makeNodeDiversityMessage(ec, bc, pnr, w, cfg)
```

File: stader/node/node.go [hunk 5]
```diff
@@ -385,12 +401,166 @@ func run(c *cli.Context) error {
...
+			if err != nil {
+				errorLog.Printlnf("Error makesNodeDiversityMessage %+v", err)
+				time.Sleep(nodeDiversityTrackerCooldown)
+
+				continue
+			}
+
+			request, err := makesNodeDiversityRequest(message, privateKey)
```
Comment (bharath-123) [inline stader/node/node.go:430]:
```suggestion
			request, err := makeNodeDiversityRequest(message, privateKey)
```

File: shared/services/ec-manager.go (hunk 3 above)
Comment (batphonghan) [reply shared/services/ec-manager.go:537]: Yea, thank you bro

**Review by bharath-123 — APPROVED** [review]

**Pull request merged**